### ISSUE TITLE
Digital signature invalid on Outlook 2017

### ISSUE THREAD

**Issue by rzagnoni** [bertjohnson/OpaqueMail#58]
Hello. We are testing OpaqueMail proxy and we have encountered a problem when we send an email with Content-Type: multipart/alternative.
using Outlook 2016 as client, when we receive an email it shows a problem with the signature. The detail of the error is: The message contents may have been altered. Signed by xxx.xxx@gmail.com using RSA/SHA1.
If we send a simple email, everything works fine but with multipart/alternative, digital signature fails.
The body of the message received is:
`From: xxx.xxx@gmail.com
To: yyy.yyy@gmail.com
Subject: TEST
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="OpaqueMail-boundary"
Content-Transfer-Encoding: 7bit

--OpaqueMail-boundary
Content-Type: text/plain
Content-Transfer-Encoding: 7bit

TEST
--OpaqueMail-boundary
Content-Type: text/html
Content-Transfer-Encoding: 7bit

TEST
--OpaqueMail-boundary--
`
we have modified OpaqueMail.TestClient to generate multipart/alternative email. If we sign email directly with TestClient without pass through OpaqueMail Proxy all works fine.

**Comment by bertjohnson** [comment]
Hi rzagnoni, thanks for the report.

I tested and found that `multipart/alternative` messages could be incorrectly tagged as `multipart/mixed`. That's fixed with release 2.5.1.

I'm not sure if that explains the issue you're seeing since I wasn't able to reproduce it. In order to test, I added the following starting at line 651 of OpaqueMail.TestClient\Form1.cs:

```cs
message.Body = "TEST";

ContentType textPlain = new ContentType();
textPlain.MediaType = "text/plain";
AlternateView av = AlternateView.CreateAlternateViewFromString("TEST", textPlain);
message.AlternateViews.Add(av);
```

That appears to work in Outlook 2016 for me. Could you please download 2.5.1 from GitHub and test again? Thanks.

**Comment by bertjohnson** [comment]
Closing as I'm unable to reproduce. Please reopen if this persists. Thanks.
